src/whylogs/app/logger.py

```diff
@@ -216,20 +278,47 @@ def log(
             return None
 
         if self.should_rotate():
-            self.rotate_time()
+            self._rotate_time()
 
+        
+        
+        #segmnet check  in case segments are just keys
+        self.profile_full_dataset=profile_full_dataset
+        if (segments is not None):
+            self.set_segments(segments)
+        
+               
+    
         if features is None and feature_name is None:
-            return
+            return 
 
         if features is not None and feature_name is not None:
             raise ValueError("Cannot specify both features and feature_name")
 
         if features is not None:
-            self._profiles[-1].track(features)
+            # full profile
+            if self.full_profile_check():
+                self._profiles[-1]["full_profile"].track(features)
+            
+            if self.segments:
+                self.log_segments(pd.DataFrame(features))
         else:
-            self._profiles[-1].track_datum(feature_name, value)
+            if (self.segments is None ) or ((self.segments is not None) and self.profile_full_dataset):
```

Comment:
extract this to a boolean with a meaningful name. I think I understand what you're trying to do but long conditions are just hard to humans to grok

Reply:
had previously defined, missed one